### PR TITLE
Fix handling of chunked+gzipped response when first chunk does not give uncompressed data

### PR DESCRIPTION
**Pull request by socketpair**
## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: aiohttp/streams.py [hunk 1]
```diff
@@ -254,15 +254,23 @@ def end_http_chunk_receiving(self) -> None:
         if self._http_chunk_splits is None:
             raise RuntimeError("Called end_chunk_receiving without calling "
                                "begin_chunk_receiving first")
-        if not self._http_chunk_splits or \
-                self._http_chunk_splits[-1] != self.total_bytes:
-            self._http_chunk_splits.append(self.total_bytes)
```
Comment (socketpair) [inline aiohttp/streams.py:259]: Previous code added zero here.

Comment (webknjaz) [reply aiohttp/streams.py:259]: Do is the whole difference in `or self.total_bytes == 0` clause?

File: aiohttp/streams.py [hunk 2]
```diff
@@ -254,15 +254,23 @@ def end_http_chunk_receiving(self) -> None:
...
 
-            # wake up readchunk when end of http chunk received
-            waiter = self._waiter
-            if waiter is not None:
-                self._waiter = None
-                set_result(waiter, False)
+        pos = self._http_chunk_splits[-1] if self._http_chunk_splits else 0
```
Comment (webknjaz) [inline aiohttp/streams.py:266]: You might want to add a comment about the meaning of `0` here.

File: tests/test_streams.py [hunk 3]
```diff
@@ -719,6 +719,34 @@ def cb():
         assert b'' == data
         assert not end_of_chunk
 
+    async def test_read_empty_chunks(self) -> None:
```
Comment (webknjaz) [inline tests/test_streams.py:722]: Could you please add a docstring here?

File: aiohttp/streams.py (hunk 1 above)
Comment (socketpair) [reply aiohttp/streams.py:259]: The problem is in adding zero as the first item of the array. Yes, total_bytes can be zero when the function is called.

**Pull request merged**